tubular/numeric.py

```diff
@@ -233,6 +233,116 @@ def transform(self, X):
         return X
 
 
+class TwoColumnOperatorTransformer(DataFrameMethodTransformer):
+
+    """
+    This transformer applies a pandas.DataFrame method to two columns (add, sub, mul, div, mod, pow).
+
+    Transformer assigns the output of the method to a new column. The method will be applied
+    in the form (column 1)operator(column 2), so order matters (if the method does not commute). It is possible to
+    supply other key word arguments to the transform method, which will be passed to the pandas.DataFrame method being called.
+
+    Parameters
+    ----------
+    pd_method_name : str
+        The name of the pandas.DataFrame method to be called.
+
+    column1_name : str
+        The name of the 1st column in the operation.
+
+    column2_name : str
+        The name of the 2nd column in the operation.
+
+    new_column_name : str
+        The name of the new column that the output is assigned to.
+
+    pd_method_kwargs : dict, default =  {'axis':0}
+        Dictionary of method kwargs to be passed to pandas.DataFrame method. Must contain an entry for axis, set to either 1 or 0.
+
+    **kwargs :
+        Arbitrary keyword arguments passed onto BaseTransformer.__init__().
+
+    Attributes
+    ----------
+    pd_method_name : str
+        The name of the pandas.DataFrame method to be called.
+
+    columns : list
+        list containing two string items: [column1_name, column2_name] The first will be operated upon by the
+        chosen pandas method using the second.
+
+    column2_name : str
+        The name of the 2nd column in the operation.
+
+    new_column_name : str
+        The name of the new column that the output is assigned to.
+
+    pd_method_kwargs : dict
+        Dictionary of method kwargs to be passed to pandas.DataFrame method.
+
+    """
+
+    def __init__(
+        self,
+        pd_method_name,
+        columns,
+        new_column_name,
+        pd_method_kwargs={"axis": 0},
+        **kwargs,
+    ):
+        """
+        Performs input checks not done in either DataFrameMethodTransformer.__init__ or BaseTransformer.__init__
+        """
+
+        if "axis" not in pd_method_kwargs.keys():
+            raise ValueError(
+                f'{self.classname()}: pd_method_kwargs must contain an entry "axis" set to 0 or 1'
+            )
+
+        if pd_method_kwargs["axis"] not in [0, 1]:
+            raise ValueError(
+                f"{self.classname()}: pd_method_kwargs 'axis' must be 0 or 1"
+            )
+
+        if not type(columns) is list:
+            if len(columns) != 2:
+                raise ValueError(
+                    f"{self.classname()}: columns must be a list containing two column names but got {columns}"
+                )
+
+        self.column1_name = columns[0]
```

Comment:
This needs a check that column1 and column2 are numeric

Reply:
added a simple check and test in [a2b37a9](https://github.com/lvgig/tubular/pull/51/commits/a2b37a971b45c5b13f2acbf9b4e5850248ccf65a)

Considering this might be a useful generic check to provide more detailed error message on (e.g. which column is not numeric, which rows etc).  An appropriate method has been developed in #57 `_check_numeric_columns` which would be good to put in a BaseNumericTransformer class and use across numeric transformers.  Intend to use this instead when available.

Reply:
I agree that we shouldn't duplicate this - for now we could just mark the method in 57 as a static method and call it from inside this transformer?

Either way I think a BaseNumericTransformer is probably a good idea

Reply:
not sure whether #57 will be ready in time for next release so going to leave this for now